client/src/pages/StockEstimation.tsx

```diff
@@ -44,6 +44,12 @@ export default function Stocks() {
     enabled: !!ticker,
   });
 
+  const [stockTimestamp, setStockTimeStamp] = useState<number>();
+
+  useEffect(() => {
+    if (state.stocks[ticker!]?.timestamp != stockTimestamp)
+      setStockTimeStamp(state.stocks[ticker!]?.timestamp);
+  });
```

Comment:
need proper side effect dep - this should fix it
```suggestion
  }, [ticker, state.stocks, stockTimestamp]);
```